### PR TITLE
ssl: fix format specifier for size_t argument to BIO_printf

### LINKED ISSUE #14519: Compile warning on mingw32 builds
```
ssl/t1_enc.c: In function 'tls1_setup_key_block':
ssl/t1_enc.c:566:50: warning: format '%ld' expects argument of type 'long int', but argument 3 has type 'size_t {aka long long unsigned int}' [-Wformat=]
         BIO_printf(trc_out, "key block length: %ld\n", num);
                                                ~~^
                                                %lld
```

### PR DESCRIPTION
**Pull request by paulidale**
Fixes #14519

### CODE REVIEW

**Review by t-j-h — APPROVED** [review]

File: ssl/t1_enc.c [hunk 1]
```diff
@@ -563,7 +563,7 @@ int tls1_setup_key_block(SSL *s)
     s->s3.tmp.key_block = p;
 
     OSSL_TRACE_BEGIN(TLS) {
-        BIO_printf(trc_out, "key block length: %ld\n", num);
+        BIO_printf(trc_out, "key block length: %zd\n", num);
```
Comment (slontis) [inline ssl/t1_enc.c:566]: I can see quite a few other places using %zu in the code..

This appears to be ANSI C99 though.

Comment (paulidale) [reply ssl/t1_enc.c]: It's out function, `BIO_printf` so the C version isn't important.
I've switched it to be unsigned.

**Review by slontis — APPROVED** [review]

**Comment by levitte** [comment]
Do note that MSVC older than 2015 doesn't know %z

**Comment by paulidale** [comment]
> Do note that MSVC older than 2015 doesn't know %z

Does MSVC throw errors if it encounters it?
Our BIO_printf knows about it.

**Comment by levitte** [comment]
> Does MSVC throw errors if it encounters it?

If compiled with `/WX` (this corresponds to gcc's `Werror`)

**Comment by paulidale** [comment]
I guess the universal fix would be to cast it to a known int type.  Not sure I like this given we've got the _z_ modifier.

**Comment by levitte** [comment]
I was a bit confounded when I mentioned MSVC, I'm currently unsure that we ask it to check for format spec mismatches, so it's possible that this will fly unnoticed.

Something to keep in mind, though, is that %z is C99, and we still proclaim that we are sticking to C90 for the language syntax, and while %z is handled by a library function (our own, to boot), we're still asking the compiler (at least gcc and clang) to check these format, which makes it a *syntax* problem.

The *proper* universal fix would have been to use PRI macros (from [<inttypes.h>](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/inttypes.h.html#tag_13_20))...  except that such macros for size_t doesn't exist.  However, that has never stopped us from rolling our own when needed, so an idea is to implement a macro PRIz with the appropriate format string (without percent).

**Comment by t8m** [comment]
However we already use %zu throughout the code.

**Comment by paulidale** [comment]
Merged to master, thanks for the feed back.

**Pull request closed without merging**